### DISCUSSION TITLE
Advised / supported hardware

### DISCUSSION THREAD

**Discussion by APAJanssen** [MannLabs/alphadia#365] [Questions / Support] [answered]
Could you perhaps share the typical hardware requirements for .raw and .d analyses? There is brief mention of 64+GB RAM in the tutorial, but nothing is stated on GPU suggestions/requirements. We would love to adopt this software, but to benchmark it against existing methods (and compare timings) we'd like to give it an honest chance to be better :)
Thanks!

**Comment by GeorgWa** [comment, marked as answer]
Hi,
I would recommend a Linux system with at least 128GB for best performance.
This should also work for variable modifications, multi-species etc. with up to 10M precursors.
Here are some other recommendations:

- GPU usage in the bundled version doesn't work reliably yet. This is mostly relevant for library prediction. Use the development version with a valid PyTorch/CUDA install like [here](https://alphadia.readthedocs.io/en/latest/installation.html#developer-installation). Hardware-wise, library prediction is rather lean. I think every NVIDIA GPU with > 8GB should be fine.
- Use Thermo or Sciex data for library-free searches. Although timsTOF analysis works well, fully predicted libraries are very slow and we have moved our focus to the Astral platform. If you want to use transfer learning with timsTOF data, I will add a guide on using alphaDIA with DIA-NN soon.
- We currently have a bug where deep-learning based FDR processing time increases significantly above 20 threads. This will hopefully be fixed soon.
- Use a fixed mass accuracy which is known to perform well. Alternatively, perform optimization (set MS1 & MS2 tolerance to 0) separately and use the consensus across a couple of runs.
- Use the two-step search as shown [here](https://alphadia.readthedocs.io/en/latest/guides/libfree-gui.html)

Feel free to reach out if you have more specific questions.
Also, reporting of reproducible bugs & performance issues is really useful for us!
Best,
Georg

**Reply by APAJanssen** [reply]
Hi Georg,

Thanks for the swift and detailed reply! We were especially wondering about the GPU usage, 64 GB VRAM would require quite an investment. Glad to read that our current hardware should be up to spec. 
I now read your comment to say that the only GPU requirement is in the library generation only, meaning the actual analysis is CPU based?

When we get everything up and running and encounter problems I'll be sure to let you know, we very much appreciate the open sourced approach you guys have taken.

Best,

Anthe

**Reply by GeorgWa** [reply]
Hi Anthe, 

yes, the search itself is mostly CPU based. Strictly speaking, the FDR deep learning training could be done on the GPU but it's fast enough on a CPU. Only the library prediction and transfer learning a GPU bound.

Looking forward to hear how it's going for your team.

Best,
Georg
